src/poliastro/earth/__init__.py

```diff
@@ -0,0 +1,142 @@
+from functools import partial
+
+from astropy import units as u
+
+from poliastro.bodies import Earth
+from poliastro.core.perturbations import J2_perturbation, atmospheric_drag_model
+from poliastro.twobody.orbit import Orbit
+from poliastro.twobody.propagation import cowell
+
+
+class EarthSatellite:
+
+    """
+    Position and velocity of a body with respect to Earth
+    at a given time
+    """
+
+    def __init__(self, orbit):
+        """Constructor.
+
+        Parameters
+        ----------
+        orbit : Orbit
+            Position and velocity of a body with respect to an attractor
+            at a given time (epoch).
+
+        Raises
+        ------
+        ValueError
+            If the orbit's attractor is not Earth
+
+        """
+
+        if orbit.attractor.name != "Earth":
+            raise ValueError("The attractor must be Earth")
+
+        self._orbit = orbit  # type: Orbit
+
+    @property
+    def orbit(self):
+        """Orbit of the EarthSatellite. """
+        return self._orbit
+
+    @classmethod
+    @u.quantity_input(r=u.m, v=u.m / u.s)
+    def from_vectors(cls, r, v):
+        """Return `Orbit` from position and velocity vectors.
+
+        Parameters
+        ----------
+        r : ~astropy.units.Quantity
+            Position vector wrt attractor center.
+        v : ~astropy.units.Quantity
+            Velocity vector.
+
+        Returns
+        -------
+        Orbit
+            New 'Orbit' position and velocity vectors.
+
+
+        """
+
+        orbit = Orbit.from_vectors(Earth, r, v)
+        return cls(orbit)
+
+    @classmethod
+    @u.quantity_input(a=u.m, ecc=u.one, inc=u.rad, raan=u.rad, argp=u.rad, nu=u.rad)
+    def from_classical(
+        cls, a, ecc, inc, raan, argp, nu,
+    ):
+        """Return ` Orbit` from classical orbital elements.
+
+        Parameters
+        ----------
+        a : ~astropy.units.Quantity
+            Semi-major axis.
+        ecc : ~astropy.units.Quantity
+            Eccentricity.
+        inc : ~astropy.units.Quantity
+            Inclination
+        raan : ~astropy.units.Quantity
+            Right ascension of the ascending node.
+        argp : ~astropy.units.Quantity
+            Argument of the pericenter.
+        nu : ~astropy.units.Quantity
+            True anomaly.
+
+        Returns
+        -------
+        Orbit
+            New 'Orbit' from classical orbital elements.
+
+        """
+
+        orbit = Orbit.from_classical(Earth, a, ecc, inc, raan, argp, nu)
+        return cls(orbit)
+
+    def propagate(self, tof, *, atmosphere=None, perturbation=None):
```

Comment:
Could we call this `gravity` again? Otherwise I think it clashes with `atmosphere`